nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/CommonClassLoaderServiceImpl.java

```diff
@@ -172,12 +172,7 @@ public String getCommonClassPath() {
     }
 
     private static String urlsToClassPath(Stream<URL> urls) {
-        return urls
-                .map(URL::getFile)
-                .filter(Predicate.not(String::isBlank))
-                .map(File::new)
-                .map(File::getAbsolutePath)
-                .collect(Collectors.joining(File.pathSeparator));
+        return urls.map(FileUtils::toFile).map(File::getAbsolutePath).collect(Collectors.joining(File.pathSeparator));
```

Comment:
Where did `.filter(Predicate.not(String::isBlank))` disappear? Why was it even there? Is it OK to remove it or it's possible that some urls are blank?
Maybe it's worth adding `.filter(File::exists)` ?

Reply:
Also note that `new File("").getAbsolutePath()` gives the value of the home directory. So I suggest adding also `.filter( f -> !f.getPath().isBlank())`

Reply:
Hmm, I see it is used by the addLibraryCommand, so you are right and that resolves even Piotrek's suggestion, because it will be longer line.  I still have doubts it is possible to get an empty string here, maybe would be useful to cover that by a test.

Reply:
`addLibraryCommand` verifies that the file exists and only then it adds the URL. So it seems that urls cannot be empty in any case and my comment is void.

Reply:
Funny, I extended tests and found much more old and dangerous bugs ... will be fixed now.